src/clients/grpc.rs

```diff
@@ -0,0 +1,310 @@
+/*
+ Copyright FMS Guardrails Orchestrator Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+use std::future::Future;
+
+#[derive(Clone)]
+pub struct GrpcClient<C> {
+    pub service_name: String,
+    pub inner: C,
+    pub tracing_enabled: bool,
+}
+
+impl<C: Clone> GrpcClient<C> {
+    pub fn new(service_name: String, inner: C, tracing_enabled: bool) -> Self {
+        Self {
+            service_name,
+            inner,
+            tracing_enabled,
+        }
+    }
+}
+
+/// Work around for compiler bug: https://github.com/rust-lang/rust/issues/96865
+pub trait SendFuture: Future {
```

Comment:
Avoids a compiler bug that seems to have existed for a couple years and has been well documented, regarding `Send` and Future dyn objects, that would otherwise occure on line 212 from:
```
let client_result: Pin<
            Box<
                dyn Future<Output = Result<tonic::Response<tonic::Streaming<_>>, tonic::Status>>
                    + Send,
            >,
        > = Box::pin($method(&mut client, $request));
```